File: app/src/main/java/fr/catch23/happyshare/ShareException.java
package fr.catch23.happyshare;

public class ShareException extends Exception {
    private String user_message = "";


    public ShareException(String user_message)
    {
        this.user_message = user_message;
    }

    public String getUser_message()
    {
        return user_message;
    }
}

File: app/src/main/java/fr/catch23/happyshare/ShareException.kt
package fr.catch23.happyshare

import android.util.Log

class ShareException(user_message: String, e: Exception = Exception()) : Exception() {
    var user_message: String
    val parent: Exception

    init {
        this.user_message = user_message
        this.parent = e

        Log.e("HappyShare", user_message)
    }
}
